src/SubscribeMe/Subscriber/MailchimpSubscriber.php

```diff
@@ -113,13 +126,98 @@ public function subscribe(string $email, array $options, array $userConsents = [
                      */
                     return true;
                 }
+                if ($body['id'] !== null) {
+                    return $body['id'];
+                }
+                return false;
             }
-
-            throw new CannotSubscribeException($exception->getMessage(), $exception);
-        } catch (RequestException $exception) {
+        } catch (ClientExceptionInterface $exception) {
             throw new CannotSubscribeException($exception->getMessage(), $exception);
         }
 
         return false;
     }
+
+    /**
+     * @see https://mailchimp.com/developer/transactional/api/messages/send-using-message-template/
+     * @inheritdoc
+     */
+    public function sendTransactionalEmail(array $emails, string|int $emailTemplateId, array $variables = []): string
+    {
+        if (empty($emails)) {
+            throw new \InvalidArgumentException('Emails information missing');
+        }
+
+        if (empty($emailTemplateId)) {
+            throw new \InvalidArgumentException('Template Id missing');
+        }
+
+        if (!is_string($this->getApiKey())) {
+            throw new ApiCredentialsException();
+        }
+
+        $valid = true;
+        foreach ($variables as $key => $value) {
+            if (!is_string($key)) {
+                $valid = false;
+                break;
+            }
+
+            if (is_array($value)) {
+                foreach ($value as $nestedKey => $nestedValue) {
+                    // @phpstan-ignore-next-line
+                    if (!is_string($nestedKey) || !(is_string($nestedValue) || is_int($nestedValue) || is_bool($nestedValue))) {
+                        $valid = false;
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (!$valid) {
+            throw new \InvalidArgumentException('Variables signature was invalid');
+        }
+
+        if (!empty($variables)) {
+            $variables = array_map(function ($key, $value) {
+                return [
+                    'name' => $key,
+                    'content' => $value
+                ];
+            }, array_keys($variables), $variables);
+        }
```

Comment:
You can extract this process to a private method `transformVariables(array $variables): array` to simplify `sendTransactionalEmail` method and better legibility